libs/StateSync-cpp/include/Microsoft/MixedReality/Sharing/StateSync/NetworkManager.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+#include <Microsoft/MixedReality/Sharing/Common/RefPtr.h>
+#include <Microsoft/MixedReality/Sharing/Common/VirtualRefCountedBase.h>
+
+#include <memory>
+#include <string>
+
+namespace Microsoft::MixedReality::Sharing::StateSync {
+
+class NetworkConnection;
+class NetworkListener;
+
+class NetworkManager : public VirtualRefCountedBase {
```

Comment:
How do I get an instance of this? Since `GetConnection()` is not static I need one, so...

Reply:
Just saw latest commit.

Reply:
These are mostly interfaces and placeholders that define the general shape.
Pushing large atomic reviews is hard, so I'm trying to do it piece by piece.

I just need this thing explicitly on C++ side to write the implementation of the RSM connections against it.

This class, in particular, will have a PInvoke monstrosity in the first iteration (for the demo). Later it will be hopefully thrown away and replaced with a lean native WebRTC wrapper.
In either case, there will be factories that create the implementations.